### PR TITLE
Modify Itinerary View

### LINKED ISSUE #17: List Available Itineraries From a Plan result
The rest of the view where the Origin/Destination UI appears should be populated with a List of itineraries that are generated when a valid origin and destination exist in the Origin/Destination UI.

## Acceptance criteria

To keep it simple for now, we'll just focus here on three core tasks.

- [ ] Automatically validate that the information in our origin/destination UI is sufficient to query the OTP server and issue the query.
- [ ] Populate the List of itineraries.
- [ ] Gracefully handle basic errors 

## mockup

Ignore "Now" and "Options" for the time being

Image

### LINKED ISSUE #38: Improve Intinerary Preview
The next step in getting OTPKit ready for public use is to improve the itinerary viewer. Right now, it's a simple list of distances and durations for each available itinerary. Let's make it easier to use and more useful by _generously borrowing_ from the Apple Maps Ux as much as possible.

image

### PR DESCRIPTION
**Pull request by hilmyveradin**
## Description

Enhance Itinerary Trip Planner View so that it becomes similar to Maps

## Related Issues

Fixes #17
Fixes #38 

## Media

image

## Test Instructions

1. Search for a place with it's trip planner
2. See if the view is changed or not

### CODE REVIEW

File: OTPKit/Features/TripPlanner/TripPlannerSheetView.swift [hunk 1]
```diff
@@ -15,41 +15,130 @@ public struct TripPlannerSheetView: View {
...
+                                    .font(.title)
+                                    .fontWeight(.bold)
+                                    .foregroundStyle(.foreground)
+                                Text(formatBusSchedule(itinerary.startTime))
+                                    .foregroundStyle(.gray)
+
+                                FlowLayout {
+                                    ForEach(Array(zip(itinerary.legs.indices, itinerary.legs)), id: \.1) { index, leg in
+                                        switch leg.mode {
+                                        case "TRAM":
```
Comment (aaronbrethorst) [inline OTPKit/Features/TripPlanner/TripPlannerSheetView.swift:69]: It looks like these could be DRY'd up into a new view called `ItineraryLegView` or something like that. You pass in `leg.route`, a background color, and an image name into the initializer.

File: OTPKit/Features/TripPlanner/TripPlannerSheetView.swift [hunk 2]
```diff
@@ -15,41 +15,130 @@ public struct TripPlannerSheetView: View {
...
+                                            }
+                                            .padding(.horizontal, 8)
+                                            .padding(.vertical, 4)
+                                            .background(Color.gray.opacity(0.2))
+                                            .foregroundStyle(.gray)
+                                            .clipShape(RoundedRectangle(cornerRadius: 8))
+                                            .frame(height: 40)
+
+                                        default:
+                                            Text("")
```
Comment (aaronbrethorst) [inline OTPKit/Features/TripPlanner/TripPlannerSheetView.swift:112]: What about other potential modes of transit? Might it be better to put something on the screen that indicates that `leg.mode` isn't something we know how to render?

**Review by aaronbrethorst — CHANGES_REQUESTED** [review]
this is looking great! I have a couple questions that I'd love to clear up before merging, but I love what you did here :)

File: OTPKit/Features/TripPlanner/TripPlannerSheetView.swift (hunk 1 above)
Comment (hilmyveradin) [reply OTPKit/Features/TripPlanner/TripPlannerSheetView.swift:69]: Thanks for the feedback Aaron! I'll try it out

File: OTPKit/Features/TripPlanner/TripPlannerSheetView.swift (hunk 2 above)
Comment (hilmyveradin) [reply OTPKit/Features/TripPlanner/TripPlannerSheetView.swift:112]: Right. That would be better. Thanks for the feedback!

**Comment by aaronbrethorst** [comment]
@hilmyveradin looking good! I pushed up some changes so you can see what I mean about DRY'ing up those leg views. Please let me know if you have any questions.

**Comment by hilmyveradin** [comment]
Ah understood. I don't have any questions right now. Thank you for improving my previous mistakes ^^

> @hilmyveradin looking good! I pushed up some changes so you can see what I mean about DRY'ing up those leg views. Please let me know if you have any questions.

**Pull request merged**